perllib/FixMyStreet/App/Form/Noise.pm

```diff
@@ -329,10 +345,41 @@ has_field source_address => (
     tags => { last_differs => 1, small => 1 },
 );
 
+has_page 'choose_location' => (
+    fields => ['possible_location_matches', 'continue'],
+    title => 'The source of the noise',
+    next => 'map',
+    update_field_list => sub {
+        my $form = shift;
+        my $saved_data = $form->saved_data;
+        my $locations = $saved_data->{possible_location_matches};
+        my $options = [];
+        for my $location ( @$locations ) {
+            push @$options, { label => $location->{address}, value => $location->{latitude} . "," . $location->{longitude} }
+        }
+        return { possible_location_matches => { options => $options } };
+    },
+    post_process => sub {
```

Comment:
process is always called on a form, both on 'arrival' and submission, so this goes a bit odd when possible_location_matches is the list, and not the single result, putting "ARRAY" in lat and null in lon. Guess it doesn't matter as is overwritten on next page, but wonder if there was a cleaner way. I used Amhurst Road as my test location.